fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Optimizer.java

```diff
@@ -245,8 +249,16 @@ private OptExpression optimizeByCost(ConnectContext connectContext,
         }
     }
 
-    private void prepare(ConnectContext connectContext, OptExpression logicOperatorTree,
-                         ColumnRefFactory columnRefFactory) {
+    private void addViewBasedPlanIntoMemo() {
+        Memo memo = context.getMemo();
+        memo.copyIn(memo.getRootGroup(), context.getLogicalTreeWithView());
+    }
+
+    private void prepare(
+            ConnectContext connectContext,
+            OptExpression logicOperatorTree,
+            ColumnRefFactory columnRefFactory,
+            ColumnRefSet requiredColumns) {
```

Comment:
Why add the `requiredColumns` into this function

Reply:
it is required to optimize LogicalPlanWithView